sdk/core/azure-core/test/ut/retry_policy.cpp

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "http/retry_policy_private.hpp"
+
+#include <gtest/gtest.h>
+
+using namespace Azure::Core::Http;
+using namespace Azure::Core::Http::Policies;
+using namespace Azure::Core::Http::Policies::_detail;
+
+TEST(RetryPolicy, Exponential)
+{
+  using namespace std::chrono_literals;
+
+  RetryOptions const options{3, 1s, 2min, {}};
+
+  {
+    std::chrono::milliseconds retryAfter{};
+    bool const shouldRetry = ShouldRetryOnTransportFailure(options, 1, retryAfter, 1.0);
```

Comment:
You're testing the internal implementation detail of the RetryPolicy.  We need to test retry policy behavior from the view of the customer.

Reply:
This is how other SDKs test it, and I thought you agreed, when you were against mocking the `Sleep()` function.

Reply:
Ho else am I supposed to test `(2^31 * MinTimeout) * RandomJitter`, without in-product test hooks for `rand()` and `sleep()`? You are giving me a contradictory requirements between code reviews.

Reply:
An existing test does smoketest retry policy with real `sleep()`: https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/core/azure-core/test/ut/policy.cpp#L97

Reply:
@RickWinter, please reply

Reply:
What I'm suggesting is different from the other case of Sleep.  Replacing Sleep and Exit would be similar to replacing ShouldRetryOnTransportFailure with a different implementation.  Replacing the implementation of the function is not what you are doing here.  
What you are doing here is testing an internal implementation detail of the public function.  I agree with you that your tests do a good job testing the internal implementation function.  However the tests do not guarantee RetryPolicy would continue to behave the same if RetryPolicy changes it internal logic.  These tests would not fail if RetryPolicy stopped calling ShouldRetryOnTransportFailure

Reply:
Ok - I added two test cases that verify that RetryPolicy does indeed call the corresponding ShouldRetry() functions. Please remove block.